core/include/jiminy/core/telemetry/TelemetryRecorder.h

```diff
@@ -29,11 +29,19 @@ namespace jiminy
 
         ////////////////////////////////////////////////////////////////////////
         /// \brief Initialize the recorder.
+        /// \param[in] telmetryData Data to log.
+        /// \param[in] timeUnit Unit with which the time will be logged
+        ///                     (note that time is logged as an int).
         ////////////////////////////////////////////////////////////////////////
-        hresult_t initialize(TelemetryData const * telemetryData);
+        hresult_t initialize(TelemetryData * telemetryData,
+                             float64_t const& timeLoggingPrecision);
 
         bool_t const & getIsInitialized(void);
 
+        /// \brief Get the maximum time that can be logged with the current precision.
+        /// \return Max time, in second.
+        float64_t getMaximumLogTime() const;
```

Comment:
`float64_t const &` for consistency, even though I agree it does really makes sense for native types.